From 9e566277bcaac9ddd0eb293ccb49b4eb27f9d401 Mon Sep 17 00:00:00 2001
From: S133PY <thiviyan@gmail.com>
Date: Sun, 30 Aug 2020 21:28:45 +0530
Subject: [PATCH] fix conflicts while compiling with rndis configs

---
 drivers/usb/gadget/function/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/function/Makefile b/drivers/usb/gadget/function/Makefile
index e04ca4b97c8e..aeba3d08e4b3 100644
--- a/drivers/usb/gadget/function/Makefile
+++ b/drivers/usb/gadget/function/Makefile
@@ -26,7 +26,7 @@ usb_f_eem-y			:= f_eem.o
 obj-$(CONFIG_USB_F_EEM)		+= usb_f_eem.o
 usb_f_ecm_subset-y		:= f_subset.o
 obj-$(CONFIG_USB_F_SUBSET)	+= usb_f_ecm_subset.o
-usb_f_rndis-y			:= f_rndis.o rndis.o
+usb_f_rndis-y			:= f_rndis.o
 obj-$(CONFIG_USB_F_RNDIS)	+= usb_f_rndis.o
 usb_f_mass_storage-y		:= f_mass_storage.o storage_common.o
 obj-$(CONFIG_USB_F_MASS_STORAGE)+= usb_f_mass_storage.o
@@ -57,7 +57,7 @@ usb_f_accessory-y               := f_accessory.o
 obj-$(CONFIG_USB_F_ACC)         += usb_f_accessory.o
 usb_f_diag-y			:= f_diag.o
 obj-$(CONFIG_USB_F_DIAG)	+= usb_f_diag.o
-usb_f_gsi-y			:= f_gsi.o rndis.o
+usb_f_gsi-y			:= f_gsi.o
 obj-$(CONFIG_USB_F_GSI)         += usb_f_gsi.o
 usb_f_cdev-y			:= f_cdev.o
 obj-$(CONFIG_USB_F_CDEV)         += usb_f_cdev.o
